src/themes/material/components/edit-chip-list/edit-chip/edit-chip.component.ts

```diff
@@ -0,0 +1,215 @@
+
+	@Input('q-grid-focus-disabled')
+	qGridFocusDisabled: boolean = true;
+
+	@Output() delete = new EventEmitter<number>();
+	@Output() add = new EventEmitter<string|any>();
+	@Output() save = new EventEmitter<string|any>();
+
+	@Output('focus-previous') focusPrevious = new EventEmitter<number>();
+	@Output('focus-next') focusNext = new EventEmitter<number>();
+
+	private shortcut = new Shortcut(new ShortcutManager());
+	private nextAction = NextAction.beginEdit;
+
+	private newValue: string;
+	private editValue: string;
+
+	isSelected: boolean = false;
+
+	get isEdited() {
+		return this.chipState === ChipState.edited;
+	}
+
+	private get isNew() {
+		return this.chipState === ChipState.new;
+	}
+
+	private get isReadOnly() {
+		return !(this.isEdited || this.isNew);
+	}
+
+	private get canMoveFocus() {
+		return this.isReadOnly && this.isSelected || this.isNew && !this.newValue;
+	}
+
+	private resetChipState() {
+		this.chipState = '';
+	}
+
+	private beginEditCommand = new Command({
+		canExecute: () => this.nextAction === NextAction.beginEdit && this.isSelected,
+		execute: () => {
+			this.editValue = this.value;
+			this.chipState = ChipState.edited;
+			this.nextAction = NextAction.endEdit;
+		},
+		shortcut: 'enter'
+	});
+
+	private endEditCommand = new Command({
+		canExecute: () => this.nextAction === NextAction.endEdit && this.editValue,
+		execute: () => {
+			this.save.emit(this.editValue);
+			this.endEdit();
+			this.editValue = '';
+		},
+		shortcut: 'enter'
+	});
+
+	private addItemCommand = new Command({
+		canExecute: () => this.nextAction === NextAction.add && this.newValue,
+		execute: () => {
+			this.add.emit(this.newValue);
+			this.endEdit();
+			this.newValue = '';
+		},
+		shortcut: 'enter'
+	});
+
+	private deleteItemCommand = new Command({
+		canExecute: () => this.isReadOnly && this.isSelected,
+		execute: () => {
+			this.delete.emit(this.index);
+		},
+		shortcut: 'backspace'
+	});
+
+	private focusNextCommand = new Command({
+		canExecute: () => this.canMoveFocus,
+		execute: () => {
+			this.focusNext.emit(this.index);
+		},
+		shortcut: 'right'
+	});
+
+	private focusPreviousCommand = new Command({
+		canExecute: () => this.canMoveFocus,
+		execute: () => {
+			this.focusPrevious.emit(this.index);
+		},
+		shortcut: 'left'
+	});
+
+	private focusPreviousFromAddNewInputCommand = new Command({
+		canExecute: () => this.canMoveFocus,
+		execute: () => {
+			this.focusPrevious.emit(this.index);
+		},
+		shortcut: 'backspace'
+	});
+
+	constructor(renderer: Renderer2, elementRef: ElementRef) {
+		super(renderer, elementRef);
+
+		const manager = new CommandManager();
+		const shortcutCommands = [
+			this.addItemCommand,
+			this.deleteItemCommand,
+			this.beginEditCommand,
+			this.endEditCommand,
+			this.focusNextCommand,
+			this.focusPreviousCommand,
+			this.focusPreviousFromAddNewInputCommand
+		];
+
+		this.shortcut.register(manager, shortcutCommands);
+	}
+
+	private deleteThis() {
```

Comment:
bad name, try to make delete(chip), This - doesn't make any sence

Reply:
Refactored this in the commit: [Merge from Dev: conflicts resolving](https://github.com/qgrid/ng2/pull/39/commits/da05f15a228a611a5059e3049c9f18402446fa8f).